modules/library/referencing/src/main/java/org/geotools/referencing/operation/projection/GeostationarySatellite.java

```diff
@@ -0,0 +1,346 @@
+package org.geotools.referencing.operation.projection;
+
+import org.geotools.geometry.DirectPosition2D;
+import org.geotools.geometry.Envelope2D;
+import org.geotools.metadata.iso.citation.Citations;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.NamedIdentifier;
+import org.opengis.parameter.*;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+import javax.measure.unit.SI;
+
+import java.awt.geom.Point2D;
+import java.util.Collection;
+
+/**
+ * The Geostationary Satellite Projection
+ * <p>
+ * Adapted from https://github.com/OSGeo/proj.4/blob/4.9/src/PJ_geos.c
+ * <p>
+ * NOTE: Not all valid coordinates in this projection will transform to valid
+ * terrestrial coordinates, this is especially true of "Full Disk" earth
+ * coverages.  If one must deal with coverages in this projection with
+ * generalized code which requires the coverage bounding-box coordinates
+ * to transform to valid terrestrial values consider clipping to a rectangle
```

Comment:
Just a note, no need to act on it, but at least rendering wise this could be achieved by writing a ProjectionHandler